src/protections/ProtectionManager.ts

```diff
@@ -70,7 +70,7 @@ export class ProtectionManager {
      */
     public async start() {
         this.mjolnir.reportManager.on("report.new", this.handleReport.bind(this));
-        this.mjolnir.matrixEmitter.on("room.event", this.handleEvent.bind(this));
+        this.mjolnir.matrixEmitter.on("room.event", (...args) => this.handleEvent(...args));
         for (const protection of PROTECTIONS) {
```

Comment:
@turt2live just double-check that this is sane? My javascript is not strong enough to be sure this is safe - the tests all pass but want to make sure I am not missing something

Reply:
looks fine, yea. It's a slightly different effect than `bind`, but accomplishes the same thing. (`=>` retains `this`)

Reply:
Bad developer hygiene: should have commented. The change is that this.handleEvent is evaluated at each emit of the event emitter so that we can install our mock from the test later.

The old version effectively copies the function and passes in this, so mutating this.handleEvent would not work.  And the `...args` is just passthrough laziness.